### ISSUE TITLE
Firebase plugin networking debug log output

### ISSUE THREAD

**Issue by kroikie** [FirebaseExtended/flutterfire#1130]
I added Firebase plugins to my flutter project and now I'm seeing this networking debug output. I'd like the option to turn it off. I have done a ton of googling and grepping and I can't seem to find out where that option is.

```
[C19.1 :64617<->172.217.25.10:443]
	Connected Path: satisfied (Path is satisfied), interface: en0
	Duration: 0.254s, DNS @0.000s took 0.018s, TCP @0.020s took 0.041s, TLS took 0.120s
	bytes in/out: 4901/1095, packets in/out: 10/9, rtt: 0.043s, retransmitted packets: 0, out-of-order packets: 0
```

**Comment by kroikie** [comment]
@lookfirst

The issue at https://github.com/flutter/flutter/issues/37148 has been closed and moved here. Future collaboration on this issue will be done here.

**Comment by kika** [comment]
Here ya go, buy me some 🍻 :-) 
https://github.com/flutter/flutter/pull/52452

Now we can wait a couple of years and it will be merged. Meanwhile, you can download the PR as a [patch](https://github.com/flutter/flutter/pull/52452.patch) and apply it to your flutter source with `git am file.patch` (if I remember correctly).

**Comment by kika** [comment]
If you don't want to wait and want to patch your current flutter installation then do the following:

```bash
# FLUTTER_HOME is where you installed flutter
cd $FLUTTER_HOME
cat <<EOF | git am - && rm $FLUTTER_HOME/bin/cache/flutter_tools.stamp && flutter doctor -v
From 64d9f0eb21a00b30e63a8b435ae579211abaa8bd Mon Sep 17 00:00:00 2001
From: Kirill Pertsev <kika@redacted.com>
Date: Wed, 11 Mar 2020 18:17:47 -0700
Subject: [PATCH] Enables log reader on iOS to (rudimentary) process multiline
 messages; Fixes #1130

---
 .../flutter_tools/lib/src/ios/simulators.dart | 12 ++++
 .../general.shard/ios/simulators_test.dart    | 56 +++++++++++++++++++
 2 files changed, 68 insertions(+)

diff --git a/packages/flutter_tools/lib/src/ios/simulators.dart b/packages/flutter_tools/lib/src/ios/simulators.dart
index 0292a5ed6d85..c71fb43071cc 100644
--- a/packages/flutter_tools/lib/src/ios/simulators.dart
+++ b/packages/flutter_tools/lib/src/ios/simulators.dart
@@ -659,6 +659,10 @@ class _IOSSimulatorLogReader extends DeviceLogReader {
 
   static final RegExp _flutterRunnerRegex = RegExp(r' FlutterRunner\[\d+\] ');
 
+  // Remember what we did with the last line, in case we need to process
+  // a multiline record
+  bool _lastLineMatched = false;
+
   String _filterDeviceLine(String string) {
     final Match match = _mapRegex.matchAsPrefix(string);
     if (match != null) {
@@ -710,6 +714,11 @@ class _IOSSimulatorLogReader extends DeviceLogReader {
       return null;
     }
 
+    // Starts with space(s) - continuation of the multiline message
+    if (RegExp(r'\s*').matchAsPrefix(string) != null && !_lastLineMatched) {
+      return null;
+    }
+
     return string;
   }
 
@@ -731,6 +740,9 @@ class _IOSSimulatorLogReader extends DeviceLogReader {
       _lastLine = _filterDeviceLine(line);
       if (_lastLine != null) {
         _linesController.add(_lastLine);
+        _lastLineMatched = true;
+      } else {
+        _lastLineMatched = false;
       }
     }
   }
EOF
```

If you see `Building flutter tool` before `flutter doctor` you are all good.

**Comment by Ehesp** [comment]
Thanks @kika - I'll close this one as it seems like a Flutter implementation rather than FlutterFire.
